<launch>
    <arg name="role_name" default="hero" />
    <arg name="control_loop_rate" default="0.05" /> 

    <node pkg="planning" type="ACC.py" name="ACC" output="screen">
        <param name="role_name" value="hero" />
        <param name="control_loop_rate" value="$(arg control_loop_rate)" />
        <rosparam file="$(find planning)/config/acc_config.yaml" command="load" />
    </node>
    <node pkg="planning" type="motion_planning.py" name="MotionPlanning" output="screen">
           <param name="role_name" value="hero" />
           <param name="control_loop_rate" value="$(arg control_loop_rate)" />
    </node>
  
    <!-- <node pkg="planning" type="dev_global_route.py" name="DevGlobalRoute" output="screen">
           <param name="from_txt" value="True" />
           <param name="sampling_resolution" value="75.0" />
           <param name="routes" value="/opt/leaderboard/data/routes_devtest.xml" />
           <param name="global_route_txt" value="/code/planning/src/global_planner/global_route.txt" />
           <param name="role_name" value="hero" />
    </node> -->
    <node pkg="planning" type="global_planner_node.py" name="PrePlanner" output="screen">
           <param name="role_name" value="hero" />
           <param name="control_loop_rate" value="1" />
           <param name="distance_spawn_to_first_wp" value="100" />
    </node>

    <node pkg="planning" type="behavior_tree.py" name="behavior_agent" output="screen">
       <rosparam file="$(find planning)/config/behavior_config.yaml" command="load" />
    </node>

    <node pkg="rqt_gui" type="rqt_gui" name="rqt_gui"/>

    <node pkg="planning" type="RequestBehaviourChangeService.py" name="RequestBehaviourChangeService" output="screen">
           <param name="role_name" value="hero" />
           <param name="control_loop_rate" value="$(arg control_loop_rate)" />
    </node>
    
</launch>
